libs/pageserver_api/src/models.rs

```diff
@@ -309,6 +310,33 @@ pub struct EvictionPolicyLayerAccessThreshold {
     pub threshold: Duration,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ThrottleConfig {
+    pub task_kinds: Vec<String>, // TaskKind
+    pub initial: usize,
+    pub interval_millis: NonZeroU64,
+    pub interval_refill: NonZeroUsize,
+    pub max: usize,
+    pub fair: bool,
+}
+
+impl ThrottleConfig {
+    pub fn disabled() -> Self {
+        Self {
+            task_kinds: vec![], // disables the rate limit
+            initial: 0,
+            interval_millis: NonZeroU64::try_from(1).unwrap(),
+            interval_refill: NonZeroUsize::try_from(1).unwrap(),
```

Comment:
Did not know these even had `try_from`, always been using `::new` because it's const. Not that `unwrap` was const.